homeassistant/components/home_connect/switch.py

```diff
@@ -19,12 +21,38 @@
     BSH_POWER_ON,
     BSH_POWER_STATE,
     DOMAIN,
+    REFRIGERATION_DISPENSER,
+    REFRIGERATION_SUPERMODEFREEZER,
+    REFRIGERATION_SUPERMODEREFRIGERATOR,
 )
-from .entity import HomeConnectEntity
+from .entity import HomeConnectDevice, HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class HomeConnectSwitchEntityDescription(SwitchEntityDescription):
+    """Switch entity description."""
+
+    on_key: str
+
+
+SWITCHES: tuple[HomeConnectSwitchEntityDescription, ...] = (
+    HomeConnectSwitchEntityDescription(
+        key="Supermode Freezer",
+        on_key=REFRIGERATION_SUPERMODEFREEZER,
+    ),
+    HomeConnectSwitchEntityDescription(
+        key="Supermode Refrigerator",
+        on_key=REFRIGERATION_SUPERMODEREFRIGERATOR,
+    ),
+    HomeConnectSwitchEntityDescription(
+        key="Dispenser Enabled",
```

Comment:
are these keys used in the unique_id? I think it would be wise to make them `snake_case`

Reply:
Yes, thought they're also used as the display Name in the UI.

Reply:
Don't they use translations keys? why did you add an icons.json file then? (as in, the keys in there correspond to translation keys)

Reply:
It looks like I accidentally deleted the translation_key in one of my commits, it's added back now.

Reply:
But does it then still use the key for the name?

Reply:
Yes, it's setup in the [parent class](https://github.com/home-assistant/core/blob/dev/homeassistant/components/home_connect/entity.py#L24).

Reply:
Uuugh, okay then we can't make it snake case